### PR TITLE
[POAE7-2308] Transformer support multi-source Cider plan

### PR DESCRIPTION
**Pull request by XinDongIntel**
### What changes were proposed in this pull request?
Change CiderPlanRewriter to support multi-source node rewriting.

### Why are the changes needed?
As V2S convertor Join support is done, change the CiderPlanRewriter to use the convertor and rewrite join pattern into  CiderPlanNode.

### Does this PR introduce _any_ user-facing change?
No.

### How was this patch tested?
Will be tested as a part of Join-related tests.

### Which label does this PR belong to?
velox-plugin

### CODE REVIEW

File: cider-velox/src/ciderTransformer/CiderPlanUtil.cpp [hunk 1]
```diff
@@ -42,7 +44,29 @@ std::shared_ptr<CiderPlanNode> CiderPlanUtil::toCiderPlanNode(
...
+  auto newRightSrc = PlanUtil::findInNodeAddrMap(newSrcMap, rightSrcBranchId, 0);
+  if (newLeftSrc.first && newRightSrc.first) {
+    auto ciderPlanNode = std::make_shared<CiderPlanNode>(planSection.target.nodePtr->id(),
+                                                         newLeftSrc.second,
+                                                         newRightSrc.second
+                                                         planSection.target.nodePtr->outputType(),
+                                                         *sPlan);
+    return ciderPlanNode;
+  } else {
+    return nullptr;
```
Comment (winningsix) [inline cider-velox/src/ciderTransformer/CiderPlanUtil.cpp:68]: Any case we have condition ```newLeftSrc.first && newRightSrc.first``` stands false? If so, should we log an error msg here?

**Review by winningsix — COMMENTED** [review]
Any chance to have some simple UTs covering this?

**Review by winningsix — DISMISSED** [review]

**Comment by winningsix** [comment]
Let's close code-style issue before merging.

File: cider-velox/src/ciderTransformer/CiderPlanUtil.cpp [hunk 2]
```diff
@@ -42,7 +44,31 @@ std::shared_ptr<CiderPlanNode> CiderPlanUtil::toCiderPlanNode(
...
+
+  std::shared_ptr<::substrait::Plan> sPlan =
+      std::make_shared<::substrait::Plan>(v2SPlanFragmentConvertor_->toSubstraitPlan(
+          planSection.target.nodePtr, planSection.source.nodePtr));
+
+  int32_t srcBranchId = planSection.source.branchId;
+  int32_t leftSrcBranchId = PlanBranches::getLeftSrcBranchId(srcBranchId);
+  int32_t rightSrcBranchId = PlanBranches::getRightSrcBranchId(srcBranchId);
+  auto newSrcMap = PlanUtil::toNodeAddrMap(srcList);
+  auto newLeftSrc = PlanUtil::findInNodeAddrMap(newSrcMap, leftSrcBranchId, 0);
```
Comment (harborn) [inline cider-velox/src/ciderTransformer/CiderPlanUtil.cpp:58]: I think a explicit type is better, do not use auto anywhere.

**Review by harborn — CHANGES_REQUESTED** [review]

**Comment by winningsix** [comment]
@XinDongIntel still has some issues around lint and code-style.

**Review by harborn — APPROVED** [review]

**Review by winningsix — APPROVED** [review]

**Pull request merged**